nuxeo-features/rest-api/nuxeo-rest-api-server/src/main/java/org/nuxeo/ecm/restapi/server/jaxrs/BatchUploadObject.java

```diff
@@ -104,6 +111,75 @@
 
     public static final String UPLOAD_TYPE_CHUNKED = "chunked";
 
+    private BatchManager batchManager;
+
+    public static class Constants {
+        public static final String MD5 = "md5";
+        public static final String MIMETYPE = "mimeType";
+        public static final String NAME = "name";
+        public static final String FILE_SIZE = "fileSize";
+        public static final String KEY = "key";
+    }
+
+    @Override protected void initialize(Object... args) {
+        super.initialize(args);
+        batchManager = Framework.getService(BatchManager.class);
+    }
+
+    @GET
+    @Path("handlers")
+    public Response providers() throws IOException {
+        Set<String> supportedHandlers = batchManager.getSupportedHandlers();
+        Map<String, Object> result = new HashMap<>();
+
+        result.put("handlers",
+                supportedHandlers
+                                 .stream()
+                                 .map(entry -> Collections.singletonMap("name",
+                                         entry))
+                                 .collect(Collectors.toList())
+        );
+
+        return buildResponse(Status.OK, result);
+    }
+
+    @GET
+    @Path("handlers/{handlerId}")
+    public Response getProviderInfo(@PathParam("handlerId") String handlerId) throws IOException {
+        BatchHandler handler = batchManager.getHandlerByName(handlerId);
+
+        if (handler == null) {
+            return Response.status(Status.NOT_FOUND).build();
+        }
+
+        Map<String, Object> result = Collections.singletonMap("name", handler.getName());
```

Comment:
What's the point if we only return the handler's name?

Reply:
I believe that this could be used to return extra meta information in the Rest API about each handler, i was hoping that this gets more fine-grained in the future.